### ISSUE TITLE
🌸 Cherry-pick request for #34639 into #2105210438002

### ISSUE THREAD

**Issue by micajuine-ho** [ampproject/amphtml#34641]
### Issue (P0 Bug)

#34639

### Pull Request(s)

#34640

### Release Tracker(s)

#34565

### Channels

Beta / Experimental

### Justification

Breaks dynamic email validation

### Verification Steps

Use this snippet:
```
<form method="post"
    action-xhr="https://example.com/subscribe"
    custom-validation-reporting="as-you-go">
    <fieldset>
      <label>
        <span>Name:</span>
        <input type="text"
          name="name"
          id="name5"
          required
          pattern="\w+\s\w+">
        <span visible-when-invalid="valueMissing"
          validation-for="name5"></span>
        <span visible-when-invalid="patternMismatch"
          validation-for="name5">
          Please enter your first and last name separated by a space (e.g. Jane Miller)
        </span>
      </label>
      <br>
      <label>
        <span>Email:</span>
        <input type="email"
          name="email"
          id="email5"
          required>
        <span visible-when-invalid="valueMissing"
          validation-for="email5"></span>
        <span visible-when-invalid="typeMismatch"
          validation-for="email5"></span>
      </label>
      <br>
      <input type="submit"
        value="Subscribe">
    </fieldset>
  </form>
```
in any format
Expected: type in "a" into the first field and red text appears showing invalid input
Broken: type in "a" into the first field and nothing happens - only on blur, red text appears showing invalid input

### Cherry-Pick Progress

To be updated by @ampproject/release-on-duty as each stage is completed.
- [x] Cherry-pick request created
- [x] Cherry-pick request approved
- [ ] New version released to Beta / Experimental channels
- [ ] New version released to Stable channel
- [ ] New version released to LTS channel
- [ ] Release tracker updated
- [ ] Cherry-pick completed

### Notifications

/cc @ampproject/release-on-duty @ampproject/wg-approvers @ampproject/cherry-pick-approvers

**Comment by newmuis** [comment]
Approved

**Comment by newmuis** [comment]
This was cherrypicked and released as part of [2105210438003](https://github.com/ampproject/amphtml/releases/2105210438003).
